holoviews/plotting/bokeh/element.py

```diff
@@ -1906,6 +1914,36 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
 
         return plot
 
+    def _apply_hard_bound(self, element, ranges):
+        """
+        Apply hard bounds to the x and y ranges of the plot.
+
+        Sets the navigable bounds of the plot based on the extents
+        of the given element and ranges. If an extend is numeric and not NaN, it is
+        used as is. Otherwise, it is set to None, which means that end of the axis
+        is unbounded.
+        """
+
+        # Skip if the element doesn't have an 'extents' attribute
+        if not hasattr(element, 'extents'):
+            return
+
+        def validate_bound(bound):
+            """Validate a single bound, returning None if it is not a valid number"""
+            return bound if isinstance(bound, (int, float)) and not np.isnan(bound) else None
```

Comment:
Would use `hv.core.util.isfinite instead here.

Reply:
replaced with 
```python
return bound if util.isfinite(bound) else None
```
since `util.isfinite` seems to handle datetimes just fine